internal/timestats/timestats.go

```diff
@@ -0,0 +1,100 @@
+package timestats
+
+import (
+	"math"
+	"sort"
+	"time"
+)
+
+type TimeStats struct {
+	Min, Max, Mean, Median, StdDev time.Duration
+	Quartiles                      [4]time.Duration
+	Deciles                        [10]time.Duration
+}
+
+func New(times []time.Duration) TimeStats {
+	n := len(times)
+	if n == 0 {
+		return TimeStats{}
+	}
+
+	// Measures computing functions works on sorted data.
+	// Sort once and compute upon the result.
+	sort.Sort(byFastest(times))
+
+	// Reused statistics measures.
+	sum := computeSum(times)
+	mean := computeMean(sum, n)
+
+	return TimeStats{
+		Min:       times[0],
+		Max:       times[len(times)-1],
+		Mean:      mean,
+		Median:    computeMedian(times),
+		StdDev:    computeStdDev(times, mean),
+		Quartiles: computeQuartiles(times),
+		Deciles:   computeDeciles(times),
+	}
+}
+
+func computeSum(values []time.Duration) time.Duration {
+	var sum time.Duration
+	for _, time := range values {
+		sum += time
+	}
+	return sum
+}
+
+func computeMean(sum time.Duration, length int) time.Duration {
+	return sum / time.Duration(length)
+}
+
+func computeMedian(sorted []time.Duration) time.Duration {
+	n := len(sorted)
+	if n%2 != 0 {
+		return sorted[n/2]
+	}
+	return computeMean(sorted[n/2-1]+sorted[n/2+1], 2)
+}
+
+func computeStdDev(values []time.Duration, mean time.Duration) time.Duration {
+	sum := time.Duration(0)
+	for _, v := range values {
+		dev := v - mean
+		sum += dev * dev
+	}
+	n := len(values)
+	return time.Duration(math.Sqrt(float64(sum / time.Duration(n))))
+}
+
+func computeDeciles(sorted []time.Duration) [10]time.Duration {
+	const nDeciles = 10
```

Comment:
For the record, `numDecile`,`numQuartile` naming refer to Go's standards (e.g. `runtime.NumGoroutine`, `reflect.Value.NumField`).
Not asking for a change though, this is perfectly fine.